compiler/rustc_resolve/src/lib.rs

```diff
@@ -973,6 +973,7 @@ pub struct Resolver<'a> {
     local_macro_def_scopes: FxHashMap<LocalDefId, Module<'a>>,
     ast_transform_scopes: FxHashMap<LocalExpnId, Module<'a>>,
     unused_macros: FxHashMap<LocalDefId, (NodeId, Ident)>,
+    unused_macro_rules: FxHashMap<(LocalDefId, usize), (NodeId, Ident)>,
```

Comment:
```suggestion
    unused_macro_rules: FxHashMap<(LocalDefId, usize), (Ident, Span)>,
```
- `NodeId` and `LocalDefId` here are the same thing so only one needs to be kept, the `NodeId` for linting can be obtained with `self.def_id_to_node_id[def_id]`
- The rule span can also be kept here if it's no longer kept in `SyntaxExtension`